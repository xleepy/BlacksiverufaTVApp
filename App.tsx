/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import { SafeAreaView, StatusBar, StyleSheet, View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Colors } from 'react-native/Libraries/NewAppScreen';
import {
  createNativeStackNavigator,
  NativeStackNavigationOptions,
} from '@react-navigation/native-stack';
import { Home } from './components/home/Home';
import { PlayerScreen } from './components/player/PlayerScreen';

const Stack = createNativeStackNavigator();

const styles = StyleSheet.create({
  navigatorContainer: {
    flex: 1,
  },
  root: {
    flex: 1,
  },
  content: {
    backgroundColor: '#161616',
  },
  header: {
    backgroundColor: '#18181b',
    boxShadow: 'rgba(0,0,0,.9) 0px 1px 2px 0px',
  },
  title: {
    color: 'white',
  },
});

const navOptions: NativeStackNavigationOptions = {
  headerStyle: styles.header,
  headerTitleStyle: styles.title,
  contentStyle: styles.content,
  headerTintColor: '#fff',
};

// docs: https://reactnavigation.org/docs/headers

function App(): JSX.Element {
  return (
    <SafeAreaView style={styles.root}>
      <StatusBar barStyle={'dark-content'} backgroundColor={Colors.darker} />
      <View style={styles.navigatorContainer}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="Home" options={navOptions} component={Home} />
            <Stack.Screen
              name="Player"
              options={navOptions}
              component={PlayerScreen}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </View>
    </SafeAreaView>
  );
}

export default App;
