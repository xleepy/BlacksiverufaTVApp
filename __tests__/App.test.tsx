import 'react-native';
import React from 'react';
import App from '../App';
import {
  render,
  fireEvent,
  screen,
  waitFor,
} from '@testing-library/react-native';

jest.mock('../components/home/hooks.ts', () => {
  return {
    useSegments: () => [
      { youtube: 'test', name: 'test', direct: 'test', id: 'test' },
    ],
  };
});

describe('App tests', () => {
  // beforeEach(() => {
  //   jest.useFakeTimers();
  // });

  // afterEach(() => {
  //   jest.runOnlyPendingTimers();
  //   jest.useRealTimers();
  // });

  it('renders correctly', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate to test', async () => {
    const { findByTestId, queryByTestId } = render(<App />);
    jest.advanceTimersByTime(0);

    const element = await findByTestId('test');
    fireEvent.press(element);

    const player = queryByTestId('test-player');

    await waitFor(() => {
      expect(player).toBeDefined();
    });

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
