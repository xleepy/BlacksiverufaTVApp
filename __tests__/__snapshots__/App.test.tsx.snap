// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenStack
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          stackPresentation="push"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backTitleVisible={true}
            backgroundColor="#18181b"
            color="#fff"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={false}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="Архив стримов BlackUFA"
            titleColor="white"
            topInsetEnabled={false}
            translucent={false}
          />
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  {
                    "backgroundColor": "#161616",
                  },
                ]
              }
            >
              <RCTSafeAreaView
                style={
                  {
                    "flex": 1,
                  }
                }
                testID="home-view"
              />
            </View>
          </View>
        </RNSScreen>
      </RNSScreenStack>
    </RNCSafeAreaProvider>
  </View>
</RCTSafeAreaView>
`;
